### PR TITLE
Add a read only embedded FileSystem implementation backed by rust-embed

### PR DESCRIPTION
**Pull request by ahouts**
Adds a read only embedded `FileSystem` implementation backed by [rust-embed](https://github.com/pyros2097/rust-embed).

The API of rust-embed is very simplistic, so if we were to implement `FileSystem.read_dir` without some kind of cache we would have to iterate through the entire directory listing every time. To prevent this I chose to use a radix trie to store the paths of files and directories when `EmbeddedFs` is constructed.

Both added dependencies are optional and gated behind a feature flag, so users who do not explicitly ask for embedded file system support will not need to download/compile them.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,4 +9,5 @@ matrix:
   allow_failures:
     - rust: nightly
     - rust: beta
-script: cargo build --verbose && cargo test --verbose
+script: cargo build --verbose && cargo test --verbose &&
+  cargo build --verbose --all-features && cargo test --verbose --all-features
```
Comment (ahouts) [inline .travis.yml:13]: Just to show the tests pass, not sure what approach you want to take for this.

File: Cargo.toml [hunk 2]
```diff
@@ -14,7 +14,11 @@ travis-ci = { repository = "manuel-woelker/rust-vfs", branch = "master" }
 
 [dependencies]
 thiserror = "1.0"
+rust-embed = { version = "5.6", optional = true }
+radix_trie = { version = "0.1", optional = true }
```
Comment (ahouts) [inline Cargo.toml:18]: Intentionally old version to support 1.32.0, the most recent version supports 1.36.0 due to a transient dependency on smallvec.

File: src/impls/embedded.rs [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+                    message: format!("{} is not a directory", path),
+                });
+            }
+            Some(&false) => (),
+        }
+
+        fn child_elements<'a>(
+            d: &'a Path,
+            t: SubTrie<'a, String, bool>,
+        ) -> Box<dyn Iterator<Item = String> + 'a> {
```
Comment (ahouts) [inline src/impls/embedded.rs:67]: If performance turns out to be an issue the dynamic iterator here could be replaced with an enum where each variant is one of the concrete types below.

**Comment by manuel-woelker** [comment]
Thanks for the PR! Should be great for fully self-contained executables.

I imagine in some cases all the paths accessed in the embedded FS are known by the application anyway, so read_dir() and the supporting trie might not even be needed.

**Pull request merged**